warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -339,203 +346,72 @@ public boolean intersect(List<JexlNode> delayedNodes) {
-            // it may be possible to reduce the tree due to the IndexMatches only coming from one side
-            // IndexMatch - A == 'a'
-            // NoData - B == 'c'
-            // IndexMatch - C == infinite
-            // the merge nodes matches now just represent (A && C)
-            if (nodeSet.size() > 1) {
-                newNode = TreeFlatteningRebuildingVisitor.flatten(JexlNodeFactory.createOrNode(nodeSet.getNodes()));
-            } else {
-                newNode = TreeFlatteningRebuildingVisitor.flatten(nodeSet.getNodes().iterator().next());
-            }
-        } else if (matches.size() == 1) {
-            newNode = TreeFlatteningRebuildingVisitor.flatten(matches.iterator().next().getNode());
-        } else {
-            JexlNodeSet nodeSet = new JexlNodeSet();
-            nodeSet.addAll(infiniteNodeSet);
-            nodeSet.add(matchNode);
-            
-            newNode = TreeFlatteningRebuildingVisitor.flatten(JexlNodeFactory.createAndNode(nodeSet.getNodes()));
-        }
-        
-        merged.myNode = newNode;
-        merged.uids = ImmutableSortedSet.copyOf(matches);
-        
-        return merged;
-    }
-    
     public IndexInfo intersect(IndexInfo o) {
         return intersect(o, new ArrayList<>(), this);
     }
     
     /**
-     * Intersection is where the logic becomes a bit more interesting. We will handle this in stages. So let's evaluate this in terms of three types of nodes <br>
-     * 1) {@code UNKNOWN -- COUNT likely > 20, but we can't be certain}<br>
-     * 2) {@code LARGE -- COUNT > 20 AT SOME POINT}<br>
-     * 3) {@code small < 20}
-     * 
-     * The reason we differentiate 1 from 2 is that 1 originates from the Index or from an Ivarat'd nodes.
-     * 
-     * Note that delayed nodes are propagated at all points.
-     * 
+     * The intersection logic is relatively straightforward as we only have three types of nodes
+     * <ol>
+     * <li>{@code INFINITE -- assume a hit exists}</li>
+     * <li>{@code LARGE -- uids > 20, were pruned and a count remains}</li>
+     * <li>{@code SMALL -- uids < 20}</li>
+     * </ol>
+     * See {@link #isInfinite()} for the definition of an infinite range.
+     * <p>
+     * We retain uids if and only if uids exist on both sides of the intersection
+     * <p>
+     * A uid and shard range will always produce a shard range. Never assume the uid hit exists on the other side.
+     * <p>
+     * In the case of two infinite ranges, we persist the -1 count for posterity
+     *
      * @param o
+     *            another {@link IndexInfo}
      * @param delayedNodes
-     * @return
+     *            a list of delayed terms
+     * @param uidIntersector
+     *            a class that helps intersect uids
+     * @return the result of an intersection operation on this IndexInfo and the other IndexInfo
      */
     public IndexInfo intersect(IndexInfo o, List<JexlNode> delayedNodes, UidIntersector uidIntersector) {
```

Comment:
We are changing the API?  Where is this being called from?  Where was this being called from?

Reply:
This isn't an API change, a few parameters were missing from the method level docs.